pykafka/producer.py

```diff
@@ -540,7 +575,40 @@ def flush(self, linger_ms, release_pending=False):
         """
         self._wait_for_flush_ready(linger_ms)
         with self.lock:
-            batch = [self.queue.pop() for _ in range(len(self.queue))]
+            batch = []
+            batch_size_in_bytes = 0
+            while len(self.queue) > 0:
+                # TODO: Should we optimistically pop and renenqueue vs. peak
```

Comment:
I suppose that's safe as long as the lock is held, but the peeking approach shouldn't be any slower and so for me, I'd prefer it the way you do it now - a bit easier to reason about.